Xamarin.Forms.Core/GestureRecognizer.cs

```diff
@@ -1,9 +1,222 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Threading.Tasks;
+
 namespace Xamarin.Forms
 {
 	public class GestureRecognizer : Element, IGestureRecognizer
 	{
+		public static readonly BindableProperty TouchStateProperty =
+			BindableProperty.Create(nameof(TouchState), typeof(TouchState), typeof(GestureRecognizer), TouchState.Default, BindingMode.OneWayToSource);
+
+		public static readonly BindableProperty TouchCountProperty =
+			BindableProperty.Create(nameof(TouchCount), typeof(int), typeof(GestureRecognizer), 0, BindingMode.OneWayToSource);
+
+		public static readonly BindableProperty TouchesProperty =
+			BindableProperty.Create(nameof(Touches), typeof(List<Touch>), typeof(GestureRecognizer), null, BindingMode.OneWayToSource);
```

Comment:
you probably want to expose a readonly collection

Reply:
so you plan to recording, and keep, ALL the touches there ? I see how it could blow-up... Shouldn't we expose a Command, or an Observer instead ?

Reply:
maybe you have ideas how to do it better? @StephaneDelcroix

Reply:
I thought of storing all touched events so that I could analyze them. it can be removed